store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStore.java

```diff
@@ -0,0 +1,349 @@
+import uk.gov.gchq.gaffer.federatedstore.operation.handler.impl.FederatedRemoveGraphHandler;
+import uk.gov.gchq.gaffer.graph.Graph;
+import uk.gov.gchq.gaffer.graph.Graph.Builder;
+import uk.gov.gchq.gaffer.jsonserialisation.JSONSerialiser;
+import uk.gov.gchq.gaffer.operation.Operation;
+import uk.gov.gchq.gaffer.operation.graph.OperationView;
+import uk.gov.gchq.gaffer.operation.impl.add.AddElements;
+import uk.gov.gchq.gaffer.operation.impl.get.GetAdjacentIds;
+import uk.gov.gchq.gaffer.operation.impl.get.GetAllElements;
+import uk.gov.gchq.gaffer.operation.impl.get.GetElements;
+import uk.gov.gchq.gaffer.operation.io.Output;
+import uk.gov.gchq.gaffer.serialisation.Serialiser;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+import uk.gov.gchq.gaffer.store.StoreException;
+import uk.gov.gchq.gaffer.store.StoreProperties;
+import uk.gov.gchq.gaffer.store.StoreTrait;
+import uk.gov.gchq.gaffer.store.operation.handler.OperationHandler;
+import uk.gov.gchq.gaffer.store.operation.handler.OutputOperationHandler;
+import uk.gov.gchq.gaffer.store.schema.Schema;
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+/**
+ * A Store that encapsulates a collection of sub-graphs
+ * and executes operations against them and returns
+ * results just like it was a single graph.
+ * <p>
+ * To create a FederatedStore you need to initialise the store with a graph name and a properties file.
+ * <p>
+ * To load a graph into the FederatedStore you need to provide three things.
+ * <ul>
+ * <li>GraphID
+ * <li>Graph Schema
+ * <li>Graph Properties file
+ * </ul>
+ * <p>
+ * To remove a graph from FederatedStore scope you need the GraphID.
+ *
+ * @see Store
+ * @see Graph
+ * @see StoreProperties
+ * @see Schema
+ */
+public class FederatedStore extends Store {
+    public static final String GAFFER_FEDERATEDSTORE = "gaffer.federatedstore.";
+    public static final String PROPERTIES = "properties";
+    public static final String SCHEMA = "schema";
+    public static final String SCHEMA_DEL_REGEX = Pattern.quote(",");
+    public static final String KEY_DEL_REGEX = Pattern.quote(".");
+    public static final String USER_IS_ATTEMPTING_TO_OVERWRITE_A_GRAPH_WITHIN_FEDERATED_STORE_GRAPH_ID_S = "User is attempting to overwrite a graph within FederatedStore. GraphId: %s";
+    public static final String GRAPH_WAS_NOT_ABLE_TO_BE_CREATED_WITH_THE_USER_SUPPLIED_PROPERTIES_GRAPH_ID_S = "Graph was not able to be created with the user supplied properties. GraphId: %s";
+    private final Map<String, Graph> graphs = Maps.newHashMap();
+    private Schema schema = new Schema();
```

Comment:
Good spot!, this is actually hiding the private super.schema I've made the super protected and deleted this schema.
Take a cookie 🍪.